cmd/oras/internal/display/track/reader.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package track
+
+import (
+	"io"
+	"os"
+	"sync/atomic"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras/cmd/oras/internal/display/progress"
+)
+
+type reader struct {
+	base         io.Reader
+	offset       atomic.Uint64
```

Comment:
What's reason to make `offset` an atomic value?
> [!Note]
> All `io.Reader`s are assumed NOT go-routine safe.

Reply:
Just trying to be go-routine safe here. Is it too much?

Reply:
Changed the implementation to non-atomic